### DISCUSSION TITLE
rez-pip and rez-test?

### DISCUSSION THREAD

**Discussion by Anti-Distinctlyminty** [AcademySoftwareFoundation/rez#1482] [Q&A] [answered]
I am writing a python library that I wish to run tests on. However, in order to install the library I first create a distributable by running `rez-env python-3.11 -- python -m build` then using `rez-pip` to install the aforementioned distributable which does a great job of installing all its dependencies. However, because this does not involve writing my own `package.py` I have no opportunity to specify the tests to be run by invoking `rez-test`.

Is this an incorrect way to develop python libraries for use in Rez?

**Comment by JeanChristopheMorinPerso** [comment]
>Is this an incorrect way to develop python libraries for use in Rez?

In my honest opinion, mixing the python packaging ecosystem with its workflows to the rez ecosystem and its workflows is not great and should be avoided. But others will have different opinions. You can't have one’s cake and eat it, too, at least not easily.

**Reply by Anti-Distinctlyminty** [reply]
That is unfortunate. Although, in this case the only thing that I'd really miss is the ability for pip to install the requirements. It's kind of a pain to have to install/release all the requirements prior to installing my own python package. Is there a way around this that I'm missing?

**Reply by JeanChristopheMorinPerso** [reply]
I would also not recommend running rez-pip inside your builds. I'm currently rewriting rez-pip and I'm planning to drop support for building packages entirely.

**Reply by Anti-Distinctlyminty** [reply, marked as answer]
No solution is perfect, but in the end I went for using rez-pip to install everything in my projects requirements.txt, then built and installed my own library during the build process using setuptools as normal.
